lib/version.js

```diff
@@ -162,36 +162,35 @@ function readLockfile (name) {
 }
 
 function updateShrinkwrap (newVersion, cb) {
-  BB.join(
+  Promise.all(
     readLockfile(SHRINKWRAP),
-    readLockfile(PKGLOCK),
-    (shrinkwrap, lockfile) => {
-      if (!shrinkwrap && !lockfile) {
-        return cb(null, false, false)
-      }
-      const file = shrinkwrap ? SHRINKWRAP : PKGLOCK
-      let data
-      let indent
-      let newline
-      try {
-        data = parseJSON(shrinkwrap || lockfile)
-        indent = detectIndent(shrinkwrap || lockfile).indent
-        newline = detectNewline(shrinkwrap || lockfile)
-      } catch (err) {
-        log.error('version', `Bad ${file} data.`)
+    readLockfile(PKGLOCK)
+  ).then(([shrinkwrap, lockfile]) => {
+    if (!shrinkwrap && !lockfile) {
+      return cb(null, false, false)
+    }
+    const file = shrinkwrap ? SHRINKWRAP : PKGLOCK
+    let data
+    let indent
+    let newline
+    try {
+      data = parseJSON(shrinkwrap || lockfile)
```

Comment:
I'd suggest http://npm.im/json-parse-even-better-errors, since we're already using it in read-package-json-fast.

I just added BOM stripping support and a `noExceptions` method to json-parse-even-better-errors, so we can replace this whole module with that one.